test/topology_custom/test_mv_topology_change.py

```diff
@@ -193,3 +194,56 @@ async def test_mv_write_to_dead_node(manager: ManagerClient):
     # will be held for long time until the write timeouts.
     # Otherwise, it is expected to complete in short time.
     await manager.remove_node(servers[0].server_id, servers[-1].server_id, timeout=30)
+
+# Reproduces https://github.com/scylladb/scylladb/issues/21492
+# In this test we perform a write with a view update while the replication factor
+# of the keyspace is being changed and the corresponding base tablet finishes
+# the migration at a different time the view tablet does.
+@pytest.mark.asyncio
+@pytest.mark.parametrize("delayed_replica", ["base", "view"])
+@pytest.mark.parametrize("altered_dc", ["dc1", "dc2"])
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_mv_rf_change(manager: ManagerClient, delayed_replica: str, altered_dc: str):
+    # The workaround for this issue depends on the fact that only last replica
+    # in the DCs replica list is modified when changing RF. Use 2 DCs to ensure
+    # that the fix works even if the new replica is not the last one in the total list.
+    servers = await manager.servers_add(2, property_file={'dc': f'dc1', 'rack': 'myrack'})
+    servers.extend(await manager.servers_add(2, property_file={'dc': f'dc2', 'rack': 'myrack'}))
+
+    cql = manager.get_cql()
+    await cql.run_async("CREATE KEYSPACE IF NOT EXISTS ks WITH replication = {'class': 'NetworkTopologyStrategy', 'dc1': 1, 'dc2': 1} AND tablets = {'initial': 1}")
+    await cql.run_async("CREATE TABLE ks.tab (pk int, ck int,  PRIMARY KEY (pk, ck))")
+    await cql.run_async("CREATE MATERIALIZED VIEW ks.tabv AS SELECT pk, ck FROM ks.tab WHERE ck IS NOT NULL PRIMARY KEY (ck, pk)")
+
+    # Insert a row so that there's data to be streamed during tablet migration
+    await cql.run_async("INSERT INTO ks.tab (pk,ck) VALUES (0,0)")
+
+    await asyncio.gather(*[manager.api.enable_injection(s.ip_addr, f"block_tablet_streaming", True, parameters={'delayed_replica': delayed_replica}) for s in servers])
+
+    ks_fut = cql.run_async(f"ALTER KEYSPACE ks WITH replication = {{'class': 'NetworkTopologyStrategy', '{altered_dc}':2}}")
+    # Perform a write after the migration has started but before it has finished
+    async def first_migration_done():
+        stages = await cql.run_async(f"SELECT stage FROM system.tablets")
+        logger.info(f"Current stages: {[row.stage for row in stages]}")
+        return set([None, "streaming"]) == set([row.stage for row in stages]) or None
```

Comment:
Is it possible to limit the SELECT only to entries related to this specific keyspace?
I'm a bit worried that maybe some other table (system table or I don't know what) gets migrated and this test will get confused.
Also, you are interested in migration of both base and view tables so don't you need to wait for both of them to start, not just one?

Reply:
> Is it possible to limit the SELECT only to entries related to this specific keyspace? I'm a bit worried that maybe some other table (system table or I don't know what) gets migrated and this test will get confused. 

Looks like it's possible. I changed it to only look at the tablets of the relevant table.

> Also, you are interested in migration of both base and view tables so don't you need to wait for both of them to start, not just one?

We actually check whether both migrations started here and one finished. Both migrations will start with `allow_write_both_read_old` and `write_both_read_old` stages, but one will hang on the `streaming` stage due to the injection, and the other one will finish with `None` migraiton stage